### PR TITLE
Fix integration test lifecycle fuctions not failing tests

### PR DESCRIPTION
**Pull request by Nokel81**
- Fail test if an async lifecycle function fails (though a bug in Jest means that the test body is still run)

Signed-off-by: Sebastian Malton <sebastian@malton.name>

part of splitting out #2003

### CODE REVIEW

File: integration/__tests__/app.tests.ts [hunk 1]
```diff
@@ -3,21 +3,22 @@ import * as utils from "../helpers/utils";
...
-
 jest.setTimeout(60000);
 
 // FIXME (!): improve / simplify all css-selectors + use [data-test-id="some-id"] (already used in some tests below)
 describe("Lens integration tests", () => {
   let app: Application;
 
   describe("app start", () => {
-    beforeAll(async () => app = await utils.appStart(), 20000);
+    beforeAll(utils.wrapJestLifecycle(async () => {
```
Comment (jim-docker) [inline integration/__tests__/app.tests.ts]: You could reduce duplication with `beforeAllWrapped(fn : () => Promise<void>, ...)`
and `afterAllTearDown()`, `afterEachTearDown()`

Comment (Nokel81) [reply integration/__tests__/app.tests.ts]: Good idea

**Review by jim-docker — APPROVED** [review]

**Review by aleksfront — APPROVED** [review]
@Nokel81 Is this bug you mentioned in description listed somewhere (my quick search not helped). I'm just thinking is it still an issue in latest jest version?

**Comment by Nokel81** [comment]
@aleksfront only the issue that I created: https://github.com/facebook/jest/issues/11060

**Pull request merged**